### ISSUE TITLE
ec2_lc assign_public_ip should default to `None`, not `False`

### ISSUE THREAD

**Issue by robbwagoner** [ansible/ansible-modules-core#273]
##### Issue Type:

Bugfix Pull Request
##### Ansible Version:

```
$ ansible --version
ansible 1.8.2 (release1.8.2 6f568bd649) last updated 2014/12/30 10:21:10 (GMT -700)
  lib/ansible/modules/core: (launch_config 3fe0c01fcb) last updated 2014/12/23 00:04:57 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD cf54dc46b4) last updated 2014/12/01 16:18:00 (GMT -700)
  v2/ansible/modules/core: (detached HEAD cb69744bce) last updated 2014/11/03 14:52:45 (GMT -700)
  v2/ansible/modules/extras: (detached HEAD 8a4f07eecd) last updated 2014/11/03 14:52:54 (GMT -700)
  configured module search path = None
```
##### Environment:

**Running From:**
- OSX 10.10 Yosemite
- Ubuntu 14.04 Trusty

**Managing:**
- Amazon Linux 2014.09
##### Summary:

Creating a AWS Launch Configuration (Ansible module `ec2_lc`) for **EC2 Classic** (not VPC) and assigning to an ASG results in AWS API Validation Error.
##### Steps To Reproduce:

**Launch Config and Auto Scaling Group**

``` yaml
- name: My Launch Configuration
  # EC2 Classic - parameter NOT used - assign_public_ip
  ec2_lc:
    name: "{{ launch_config_name }}"
    image_id: '{{ image_id }}'
    instance_type: '{{ instance_type }}'
    key_name: '{{ key_name }}'
    region: '{{ region }}'
    security_groups: '{{ security_groups }}'
    user_data: '{{ user_data }}'
    instance_monitoring: yes
  register: launch_config

- name: Auto Scaling Group
  # EC2 Classic - parameter NOT used - vpc_zone_identifier
  ec2_asg:
    name: "{{ asg_name }}"
    region: "{{ region }}"
    availability_zones: "{{ availability_zones }}"
    load_balancers: "{{ load_balancers }}"
    health_check_type: ELB
    health_check_period: 420
    launch_config_name: "{{ launch_config_name }}"
    replace_all_instances: yes
    replace_batch_size: 1
    region: "{{ region }}"
    availability_zones: "{{ availability_zones }}"
    load_balancers: "{{ load_balancers }}"
    health_check_type: ELB
    health_check_period: 420
    launch_config_name: "{{ launch_config_name }}"
```
##### Expected Results:

The ASG is updated with the assigned Launch Config.
##### Actual Results:

``` XML
TASK: [autoscaling | Auto Scaling Group] ************************************** 
<localhost> REMOTE_MODULE ec2_asg max_size=1 launch_config_name='staging sclera/release-v0.7.1 iris_api/v0.99.8 ansible-iris/40cad98' health_check_type=ELB state=present desired_capacity=1 min_size=1 region=us-east-1 name=continuous-integration-staging
failed: [localhost] => {"failed": true}
msg: BotoServerError: 400 Bad Request
<ErrorResponse xmlns="http://autoscaling.amazonaws.com/doc/2011-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>AssociatePublicIpAddress field can not be specified without a subnet id</Message>
  </Error>
  <RequestId>c233e9eb-6479-11e4-80bc-5130fbe3ad32</RequestId>
</ErrorResponse>
```
###### Boto LaunchConfiguration object

[boto.ec2.autoscale.launchconfig.LaunchConfiguration](https://github.com/boto/boto/blob/develop/boto/ec2/autoscale/launchconfig.py#L98-L106) class defaults  `associate_public_ip_address` to `None`. The `ec2_lc` Ansible module defaults `associate_public_ip_address` to `False`. Which means, the instances are in a VPC, but should not get a public IP assigned.

| STATES |  |
| --- | --- |
| `None` | EC2 Classic |
| `False` | In VPC, do not assign public IP |
| `True` | in VPC, assign public IP |

**Comment by gdamjan** [comment]
This needs to go in ASAP. ec2_lc is almost useless now (almost == for everyone not using a VPC).

**Comment by robbwagoner** [comment]
Reformatted the bug report to match the [Ansible Issue template](https://raw.githubusercontent.com/ansible/ansible/devel/ISSUE_TEMPLATE.md) with the hope it will increase the likelihood of my bugfix PR being accepted. Thanks!

**Comment by bcoca** [comment]
closed by merge of #105
